workstreams/Adaptive_Profile_Workstream.md

```diff
@@ -0,0 +1,160 @@
+outlining the use of data types and values should define any cases of undefined
+behavior and/or cases that result in failure (only) during execution, as it may
+be the case for example when dividing by zero. It is a non-goal to be able to
+ensure that classical values remain within the supported range for that data
+type as part of compilation or validation. The profile should define a mechanism
+for handling failures at runtime that does not require runtime support beyond
+what is otherwise needed for program execution. The mechanism specifically
+should not require aborting execution upon failure or supporting logging a
+string value during execution.
+
+Any Base Profile compliant program should also be compliant with the Adaptive
+Profile. If revisions to the full specifications or to the Base Profile
+specification are required or beneficial to meet this requirement they should be
+raised as early as possible with the steering committee.
+
+## Dependencies & Related Projects
+
+A related workstream is to specify entry point functions including runtime
+initialization and finalization in the QIR specification. Any rules for entry
+points defined as part of this profile should be compliant with that
+specification. The specification of an Adaptive Profile should also naturally
+extend the Base Profile specification that has already been defined.
+
+## Deliverable(s) & Expected Outcome
+
+The expected outcome is a document that clearly defines an Adaptive Profile
+meeting the requirements outlined [above](#requirements). The intent is for the
+Adaptive Profile to permit expressing quantum programs that contain classical
+computations with certain data types and non-trivial control flow structures. It
+will limit expressiveness to guarantee that the program always terminates (no
+unbounded loops or recursions). This specification will replace what is
+currently drafted as [Profile
+B](https://github.com/qir-alliance/qir-spec/blob/main/specification/v0.1/7_Profiles.md#profile-a-basic-quantum-functionality)
+on the [specification repository](https://github.com/qir-alliance/qir-spec).
+
+Like the Base Profile, the Adaptive Profile should not rely on any particular
+quantum instruction(s) being available. To target a quantum program to a
+specific backend requires selecting a suitable profile and a compatible quantum
+instruction set. The profile and the quantum instruction set (QIS) selection
+together determine how the program IR should be represented.
+
+A similar specification for the Base Profile can be found
+[here](https://github.com/qir-alliance/qir-spec/blob/main/specification/under_development/profiles/Base_Profile.md).
+The specification of the Adaptive Profile should follow a similar outline for
+the sake of consistency and comprehensiveness of the specification.
+
+## Future Work (out of scope for the Adaptive Profile)
+
+The following features will not be supported in programs that are compliant with
+the Base Profile:
```

Comment:
Should this read `compliant with the Adaptive Profile`?

Reply:
Yes, thanks.